src/main/java/cz/xtf/wait/Waiter.java

```diff
@@ -0,0 +1,133 @@
+package cz.xtf.wait;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * An object that waits on condition to be met.
+ */
+public interface Waiter {
+	long DEFAULT_INTERVAL = 1_000L;
+	long DEFAULT_TIMEOUT = 60_000;
+
+	/**
+	 * Sets waiter timeout after which waiters stops waiting. Timeout is logged
+	 * in case of setting {@link Waiter#logPoint(LogPoint)} to anything else then
+	 * {@link LogPoint#NONE}
+	 *
+	 * @param millis timeout in milliseconds
+	 * @return this
+	 */
+	Waiter timeout(long millis);
+
+	/**
+	 * Sets waiter timeout after which waiters stops waiting. Timeout is logged
+	 * in case of setting {@link Waiter#logPoint(LogPoint)} to anything else then
+	 * {@link LogPoint#NONE}
+	 *
+	 * @param timeUnit timeUnit that is converts time {@code t} to milliseconds
+	 * @param t timeout in {@code timeUnit}
+	 * @return this
+	 */
+	Waiter timeout(TimeUnit timeUnit, long t);
+
+	/**
+	 * Sets waiter conditions check interval.
+	 *
+	 * @param millis interval in milliseconds
+	 * @return this
+	 */
+	Waiter interval(long millis);
+
+	/**
+	 * Sets waiter conditions check interval.
+	 *
+	 * @param timeUnit timeUnit that is converts time {@code t} to milliseconds
+	 * @param t interval in milliseconds
+	 * @return this
+	 */
+	Waiter interval(TimeUnit timeUnit, long t);
+
+	/**
+	 * Sets waiting reason.
+	 *
+	 * @param reason what the waiters what upon.
+	 * @return this
+	 */
+	Waiter reason(String reason);
+
+	/**
+	 * Sets waiters logPoints.
+	 * @see LogPoint
+	 *
+	 * @param logPoint what points of waiting should be logged.
+	 * @return this
+	 */
+	Waiter logPoint(LogPoint logPoint);
+
+	/**
+	 * Waits till condition is met.
+	 *
+	 * @return true if wanted condition was met, false if unwanted state condition was met
+	 * @throws TimeoutException in case that neither wanted nor unwated condition have been met
+	 * @throws InterruptedException by {@link Thread#sleep}
+	 */
+	boolean run() throws TimeoutException, InterruptedException;
```

Comment:
I'm not exactly sure by naming this method. 'wait' would be ideal but that is not an option because of Object.wait(). Another options that came to my mind are: doWait, waitFor, run, execute or start. Ideas?

Reply:
My subjective priority list would be `execute(), doWait(), perform()`.

Reply:
And maybe I'd like to see `InterruptedException` wrapped with `TimeoutException` for simplifying the declaration. The `catch{}` block was usually shared for both of those, or at least I've done it.:)

Reply:
Wrapping `InterruptedException` seems reasonable.

Reply:
renamed method to `execute` and swallowed up `InterruptedException`. Note that `assertEventually` is also present if you want to swallow up also TimeoutException.